### ISSUE TITLE
UseRoot Causes double re-render

### ISSUE THREAD

**Issue by vinceprofeta** [Yomguithereal/baobab-react#134]
When using `useRoot` my app component renders twice. Using the higher order root resolves the issue.

**Comment by Yomguithereal** [comment]
Hello @vinceprofeta. Would you be able to open a PR to solve this issue?

**Comment by vcardins** [comment]
Any update on this?

**Comment by Yomguithereal** [comment]
No. Can you provide a minimal reproducible example so I can investigate?

**Comment by georgehenderson** [comment]
The issue seems to be centered around this effect. It causes the component tree to render when it sets the tree state inside of the effect. You essentially set state twice . https://github.com/Yomguithereal/baobab-react/blob/master/src/hooks.js#L31

I’m not entirely sure it’s needed using the tree as a singleton. My workaround was to just use a higher-order component for the root like Vince said above.
